server/data.js

```diff
@@ -0,0 +1,249 @@
+    keywords: ['dancing', 'ballroom', 'ballet', 'story', 'art'],
+  },
+  {
+    id: 23,
+    title: 'People Dancing :: Home :: Foundation for Community Dance',
+    url: 'https://www.communitydance.org.uk/about-people-dancing',
+    desc:
+      'People Dancing is the development and membership organisation for community and participatory dance, driving forward inclusion and excellence through professional development',
+    keywords: ['dancing', 'ballroom', 'ballet', 'story', 'art'],
+  },
+  {
+    id: 24,
+    title: 'Dancesportinfo.net',
+    url: 'https://www.dancesportinfo.net/',
+    desc:
+      'Everything dancesport - information, competition timetable, dance practice, partner search',
+    keywords: ['dancing', 'ballroom', 'ballet', 'story', 'art'],
+  },
+  {
+    id: 25,
+    title: 'All about DanceSport at World DanceSport Federation on ...',
+    url: 'https://www.worlddancesport.org/',
+    desc:
+      "Record Numbers Attend Online Adjudicators' Congress · The competition calendar saw some daylight · Hall of Fame to Celebrate Century of DanceSport · WDSF ...",
+    keywords: ['dancing', 'ballroom', 'ballet', 'story', 'art'],
+  },
+  {
+    id: 26,
+    title: 'World Dance Council',
+    url: 'https://www.wdcdance.com/',
+    desc:
+      'The WDC is the world authority for Professional Dancing incorporating the World Competitive Dance Committee and the World Social Dance Committee.',
+    keywords: ['dancing', 'ballroom', 'ballet', 'story', 'art'],
+  },
+  {
+    id: 27,
+    title: 'Red Bull BC One - B-Boy and B-Girl competition',
+    url: 'https://www.redbull.com/gb-en/event-series/bc-one',
+    desc:
+      'Red Bull BC One is the biggest and most prestigious one-on-one breaking competition in the world. Every year, thousands of breakers compete for a chance to represent at the Red Bull BC One World Final.',
+    keywords: ['dancing', 'ballroom', 'ballet', 'story', 'art'],
+  },
+  {
+    id: 28,
+    title: 'Alexander Whitley',
+    url: 'https://www.alexanderwhitley.com/',
+    desc: 'Website of choreographer Alexander Whitley',
+    keywords: ['dancing', 'ballroom', 'ballet', 'story', 'art'],
+  },
+  {
+    id: 29,
+    title: "Sadler's Wells Theatre - London's Dance House - London",
+    url: 'https://www.sadlerswells.com/',
+    desc:
+      "The world's No.1 venue dedicated to international dance - presenting dance in all forms from contemporary to flamenco, ballet to hip hop and more.",
+    keywords: ['dancing', 'ballroom', 'ballet', 'story', 'art'],
+  },
+  {
+    id: 30,
+    title: 'Greenwich Dance - London',
+    url: 'https://greenwichdance.org.uk/',
+    desc:
+      'Greenwich Dance is the home of dance in South East London, hosting classes, workshops, events and performances across the Royal Borough of Greenwich.',
+    keywords: ['dancing', 'ballroom', 'ballet', 'story', 'art'],
+  },
+];
+
```

Comment:
We think you've written logic to filter out the relevant keywords in your data, which is fine!

However, we suspect it might be easier to have tailored keyword arrays for each website entry in your database e.g. Object 2 could just have football related keywords in your keyword array.  This may be contributing to unrelated search results appearing for the cooking search.